### PR TITLE
Padding

### PR DESCRIPTION
**Pull request by marcmengel**
This is 
* changes to support padding  (--padding to make_spack, make_subspack, script fixes to handle padding)
* using getopt to improve arg parsing in most scripts
* handling multiple externals in /cvmfs of packages in make_packages_yaml
* adding making module files to declare_simple
* scd_recpies packages (htcondor, cigetcert, etc) checked for in make_packages_yaml

### CODE REVIEW

**Comment by greenc-FNAL** [comment]
Sorry, just saw this. Reviewing now...

File: bin/make_packages_yaml [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+force_x86_64="(.*proto|asciidoc|autoconf|autogen|automake|binutils|bison|bzip2|cmake|conv|curl|damageproto|davix|diffutils|environment-modules|expat|findutils|fixesproto|flex|font-util|fontconfig|font-util|libfontenc|libxfont|mkfontdir|mkfontscale|freeglut|freetype|gdb|gdbm|gettext|git|gl2ps|glew|gmake|help2man|icu4c|jemalloc|libbsd|libdrm|libedit|libevent|libffi|libice|libiconv|libjpeg|libjpeg-turbo|libgcrypt|libpciaccess|libpng|libpthread-stubs|libsigsegv|libsm|libtool|libuuid|libx11|libxau|libxcb|libxdamage|libxdmcp|libxext|libxfixes|libxft|libxi|libxmu|libxpm|libxrandr|libxrender|libxshmfence|libxt|libxv|libxvmc|libxxf86vm|libyaml|lua|lz4|m4|matplotlib|mesa|mesa-glu|mpich|mysql|nasm|ncurses|openblas|openssl|patchelf|pcre|pcre2|perl|pkgconf|pkg-config|readline|sqlite|swig|tar|tcl|texinfo|texlive|tk|unzip|util-macros|xxhash|xz|zlib)"
```
Comment (greenc-FNAL) [inline bin/make_packages_yaml]: `.*proto` redundant with `damageproto`, `fixesproto`?

**Review by greenc-FNAL — APPROVED** [review]
Looks good. Didn't go into too much detail because brain foggy today.

For future consideration:

* `make_spack` to take options on repos for Spack and spack-infrastructure in addition to branch/tag.
* Remove hwloc: presence is rare enough and it's simple enough to build that we should just use Spack's version unconditionally: will increase likely compatibility between Jenkins-built binaries and external nodes.

File: bin/make_packages_yaml (hunk 1 above)
Comment (marcmengel) [reply bin/make_packages_yaml]: Thanks for noticing that; that was a merge mixup from the rebasing...
Also removing hwloc from the packagelist.

**Pull request merged**